channeld/channeld.c

```diff
@@ -3450,6 +3450,39 @@ static struct inflight *inflights_new(struct peer *peer)
 	return inf;
 }
 
+static void update_hsmd_with_splice(struct peer *peer, struct inflight *inflight)
+{
+	u8 *msg;
+
+	// These aren't allowed to change, so we don't need to gather them
+	u8 *local_upfront_shutdown_script = NULL;
+	u32 * local_upfront_shutdown_wallet_index = NULL;
+	u8 *remote_upfront_shutdown_script = NULL;
+
+	// FIXME - the push_value needs to reflect what is owed to the non-opener
+	struct amount_msat push_value = AMOUNT_MSAT(0);
+
+	msg = towire_hsmd_ready_channel(
+		NULL,
+		peer->channel->opener == LOCAL,
+		inflight->amnt,
+		push_value,
+		&inflight->outpoint.txid,
+		inflight->outpoint.n,
+		peer->channel->config[LOCAL].to_self_delay,
+		local_upfront_shutdown_script,
+		local_upfront_shutdown_wallet_index,
+		&peer->channel->basepoints[REMOTE],
+		&peer->channel->funding_pubkey[REMOTE],
```

Comment:
the funding pubkey is allowed to vary in the splice, and we need to be involved in that.  I'm wondering where that happens and how we will be informed of the remote one (and asked about the local one).

Reply:
> we need to be involved in that

Are you saying that roundtrips/protocol is missing?

> how we will be informed of the remote one

I think we are being informed with this call.  Am I missing something?

Reply:
> funding pubkey is allowed to vary in the splice

https://github.com/lightning-signer/c-lightning/blob/2023-08-explore-splicing/channeld/channeld.c#L3536-L3539

That seems to imply it is not changing?

Reply:
This is where we are telling the other side our key:

https://github.com/lightning-signer/c-lightning/blob/2023-08-explore-splicing/channeld/channeld.c#L3548-L3552

Reply:
each of the prior is one side, presume there are similar in the other "direction"

Reply:
the spec allows for it to change (for the whole splice, not per RBF candidate).  we should ask the CLN team about this.

Reply:
I just scoured the spec looking for where it is allowed to change ... can you point me to it?

Reply:
https://github.com/lightning/bolts/pull/863/files#diff-ed04ca2c673fd6aabde69389511fa9ee60cb44d6b2ef6c88b549ffaa753d6afeR522

```
    type: 74 (splice)

    data:
        [chain_hash:chain_hash]
        [channel_id:channel_id]
        [u64:funding_satoshis]
        [u32:funding_feerate_perkw]
        [u32:locktime]
        [point:funding_pubkey]

    type: 76 (splice_ack)

    data:
        [chain_hash:chain_hash]
        [channel_id:channel_id]
        [u64:funding_satoshis]
        [point:funding_pubkey]
```

Reply:
looks like CLN explicitly doesn't support changing them, but I assume that will be fixed before release.  need to ask them.